apps/sms/services/js/activity/client.js

```diff
@@ -0,0 +1,85 @@
+/* global EventDispatcher,
+          bridge
+*/
+
+(function(exports) {
+  'use strict';
+
+  /**
+   * Name of the service that is responsible for managing activities.
+   * @type {string}
+   */
+  const SERVICE_NAME = 'activity-service';
+
+  /**
+   * Reference to active bridge client instance.
+   * @type {Client}
+   */
+  var client = null;
+
+  /**
+   * Indicates if there is pending activity request.
+   * @type {boolean}
+   */
+  var hasPendingRequest = false;
+
+  var ActivityClient = {
+    /**
+     * Initialized activity service client bridge.
+     */
+    init() {
+      client = bridge.client(SERVICE_NAME);
+
+      client.on('activity-request', (request) => {
+        hasPendingRequest = true;
```

Comment:
This works because our activities are all inline -- but maybe you could postError automatically if hasPendingRequest is true already ? Or you prefer that hasPendingRequest is just a way to know that an activity is happening ?